### ISSUE TITLE
Build snaps for architectures other than x86_64

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#7674]
Adrien has a branch where he had this working using QEMU.

**Comment by maximillianh** [comment]
@adferrand that's great! Can you share the branch with us?

**Comment by adferrand** [comment]
Of course! It the branch `docker-multiarch` on my fork https://github.com/adferrand/certbot-snap-build. I need to sync with the latest modifications from upstream however.

**Comment by basak** [comment]
See also https://github.com/basak/certbot-snap-build/issues/7 which looks promising: Travis CI is [now supporting](https://blog.travis-ci.com/2019-10-07-multi-cpu-architecture-support) ARM.

**Comment by adferrand** [comment]
Yes, I studied this solution. In term of performance and simplicity, it is really good. However, choosing this solution implies that the following drawbacks are acceptable:
1) The feature is yound (september 2019) and in alpha stage. So it can change significantly in the future. Also, it relies on LXD for which is [officially unsupported](https://askubuntu.com/questions/1060089/lxd-running-an-image-with-a-foreign-architecture)
2) Only arm64 is supported, no 32 bits ARM CPUs (eg. armhf, that is widely used in the Raspberry ecosystem)
3) Multi-arch support for Certbot snap would be strongly tied to Travis: no other CI proposes that, and it is also not buildable locally.

I am working on a Docker + QEMU approach. It is better supported by the community, but significantly slower during the build. I am mitigating that by exposing to pip locally pre-built wheels of cryptography and cffi for the target architectures.

**Comment by bmw** [comment]
When this feature lands and has been included in a snap pushed to the beta channel, we should update the instruction generator to remove the warning I added in https://github.com/certbot/website/pull/558 saying that our snaps only support x86_64.
